src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,58 @@
+package duke;
+
+import duke.Command.*;
+
+public class Parser {
+
+    /**
+     * Parses the input provided into relevant Commands
+     *
+     * @param input String of next line of user input
+     * @return Relevant Command corresponding to input
+     * @throws DukeException
+     */
+    public static Command parse(String input) throws DukeException{
+        String first_word = input.split(" ")[0];
+
+
+        if (input.equals("bye")) {
+            return new ExitCommand();
+        } else if (input.equals("list")) {
+            return new ListCommand();
+        } else if (first_word.equals("done")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.split(" ")[1]) - 1;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter an integer after 'done'. (e.g. done 2)");
+            }
+            return new DoneCommand(index);
+        } else if (first_word.equals("delete")) {
+            int index;
+            try {
+                index = Integer.parseInt(input.split(" ")[1]) - 1;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter an integer after 'delete'. (e.g. delete 2)");
+            }
+            return new DeleteCommand(index);
+        } else if (first_word.equals("find")){
+            String remaining;
+            try {
+                remaining = input.split(" ", 2)[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, please enter a keyword after 'find'.");
+            }
+            return new FindCommand(remaining);
+        } else {
+            String remaining;
+            try {
+                remaining = input.split(" ", 2)[1];
+            } catch (ArrayIndexOutOfBoundsException e) {
+                throw new DukeException("Sorry, tasks must include descriptions.");
+            }
+            return new AddCommand(first_word, remaining);
+        }
```

Comment:
Perhaps we can use `enum` here to simplify the logic here! Also, personally I fell that all the `try-catch` statement here can be abstracted to other class method e.g. `createDone()` etc. to prevent arrowhead style code, and that one single `catch` for `ArrayIndexOutOfBoundsException` could be used for the entire method. Don't know if I articulate my idea in a very accurate way 😂